[include]
files = supervisord.base.conf

; supervisorctl complains if it doesn't find itself and doesn't check includes
[supervisorctl]
serverurl = unix:///var/run/supervisor.sock

[fcgi-program:uvicorn]
; expose django port to the network (but this does not expose outside the docker host)
socket = tcp://0.0.0.0:8000
# command = sh -c 'exec timeout -v --kill-after=$((%(ENV_UVICORN_NUM_PROCS)s * 150 + 10)) $((%(ENV_UVICORN_NUM_PROCS)s * 600 + 10)) uvicorn --fd 0 --timeout-graceful-shutdown 5 --limit-concurrency 500 --limit-max-requests 2000 --backlog 200 --timeout-keep-alive 5 tph.asgi:application'
command = uvicorn --fd=0 --workers=1 --limit-concurrency=100 --backlog=50 --timeout-keep-alive=5 --timeout-graceful-shutdown=5 --ws-max-queue=20 --ws-ping-interval=5 --ws-ping-timeout=5 --proxy-headers --forwarded-allow-ips="*" --loop="asyncio" tph.asgi:application
directory = /app/server
numprocs = %(ENV_UVICORN_NUM_PROCS)s
process_name = uvicorn-%(process_num)d
autostart = %(ENV_ENABLE_BACKEND)s
autorestart = true
stdout_logfile = /srv/logs/uvicorn.log.ignorethese
stdout_logfile_maxbytes = 999999
stdout_logfile_backups = 10
stderr_logfile = /srv/logs/uvicorn.debug.log.ignorethese
stderr_logfile_maxbytes = 999999
stderr_logfile_backups = 10

; periodically bounce uvicorn processes to prevent lockup
; disabled because the locked processes hang around?
# [program:periodic-restart]
# command = bash -c 'for ((i=0; i<(%(ENV_UVICORN_NUM_PROCS)s); i++)); do sleep 150; echo "Periodic restart: uvicorn-$i"; echo "restart uvicorn:uvicorn-$i" | supervisorctl; done'
# autostart = %(ENV_ENABLE_BACKEND)s
# autorestart = true
# stdout_logfile = /srv/logs/periodic-restart.log
# stdout_logfile_maxbytes = 999999
# stdout_logfile_backups = 10
# stderr_logfile = /srv/logs/periodic-restart.log
# stderr_logfile_maxbytes = 999999
# stderr_logfile_backups = 10

; autorestart is false because this should only fail on login errors, which we
; don't want to spam
[program:watch-email]
command = /app/server/scripts/watch_email.py
directory = /
environment = PYTHONPATH="/app/server/"
autostart = %(ENV_ENABLE_BACKEND)s
autorestart = false
stdout_logfile = /srv/logs/watch_email.log
stdout_logfile_maxbytes = 999999
stdout_logfile_backups = 10
stderr_logfile = /srv/logs/watch_email.log
stderr_logfile_maxbytes = 999999
stderr_logfile_backups = 10

[program:discordhintreact]
command = /app/server/scripts/discord_bot.py
directory = /
environment = PYTHONPATH="/app/server/"
autostart = %(ENV_ENABLE_BACKEND)s
# Not set up yet, autorestart once ready
autorestart = true
stdout_logfile = /srv/logs/discord_hint_react.log
stdout_logfile_maxbytes = 999999
stdout_logfile_backups = 10
stderr_logfile = /srv/logs/discord_hint_react.log
stderr_logfile_maxbytes = 999999
stderr_logfile_backups = 10

; serve fixtures on test_branch deploys but on a port only visible within the docker network
[program:serve-fixtures]
command = caddy file-server --listen :8765 --root /app/server/tph/fixtures
directory = /app/server
autostart = %(ENV_SERVE_FIXTURES)s
autorestart = false
stdout_logfile = /srv/logs/serve_fixtures.log
stdout_logfile_maxbytes = 999999
stdout_logfile_backups = 10
stderr_logfile = /srv/logs/serve_fixtures.log
stderr_logfile_maxbytes = 999999
stderr_logfile_backups = 10
